centreon/www/class/centreonAuth.LDAP.class.php

```diff
@@ -254,18 +244,22 @@ public function updateUserDn(): bool
             }
             //getting user's pager
             $userPager = $this->contactInfos['contact_pager'];
-            if (
-                isset($userInfos[$this->ldap->getAttrName('user', 'pager')])
-                && trim($userInfos[$this->ldap->getAttrName('user', 'pager')]) != ''
-            ) {
-                if (is_array($userInfos[$this->ldap->getAttrName('user', 'pager')])) {
-                    // Get the first if there are multiple entries
-                    if ($userInfos[$this->ldap->getAttrName('user', 'pager')][0]) {
-                        $userPager = $userInfos[$this->ldap->getAttrName('user', 'pager')][0];
-                    }
-                } elseif ($userInfos[$this->ldap->getAttrName('user', 'pager')]) {
-                    $userPager = $userInfos[$this->ldap->getAttrName('user', 'pager')];
+            try {
+                $ldapUserPager = $userInfos[$this->ldap->getAttrName('user', 'pager')];
+                if (isset($ldapUserPager)) {
+                    $userPager = (trim(is_array($ldapUserPager) ? current($ldapUserPager) : $ldapUserPager));
+                }
+
+                if ($userPager === '') {
+                    throw new Exception('User pager must be a non-empty string');
                 }
+            } catch (Exception $ex) {
+                $this->CentreonLog->insertLog(
+                    3,
+                    'LDAP AUTH - Error : Invalid user pager : ' . $ex->getMessage()
+                );
+
+                return false;
             }
```

Comment:
My 2cts
- try catch block seems a bit overkill and add unnecessary cognitive complexity for the single Exception you throw yourself IMHO
- an enhanced logger has been provided: please take the opportunity when you change some code to use it

```suggestion
            $ldapUserPager = $userInfos[$this->ldap->getAttrName('user', 'pager')];
            if (isset($ldapUserPager)) {
                $userPager = (trim(is_array($ldapUserPager) ? current($ldapUserPager) : $ldapUserPager));
            }

            if ($userPager === '') {
                CentreonLog::create()->error(
                    logTypeId: CentreonLog::TYPE_LDAP,
                    message: 'LDAP AUTH - Error : Invalid user pager : User pager must be a non-empty string',
                );

                return false;
            }
```

The same goes for the similar change above.
@Drago37 WDYT?

Reply:
Beware, here it's CentreonUserLog that is used, not CentreonLog :) I think we would deprecate CentreonUserLog to use CentreonLog. No ? What are the differences ? I prefere to start a request change to discuss before merging the PR.

Reply:
It's ok for me ;) seen together !